lib/fetch/request.js

```diff
@@ -406,26 +406,26 @@ class Request {
     if (Object.keys(init).length !== 0) {
       // 1. Let headers be a copy of this’s headers and its associated header
       // list.
-      let headers = new Headers(this.headers)
+      let headers = new Headers(this[kHeaders])
 
       // 2. If init["headers"] exists, then set headers to init["headers"].
       if (init.headers !== undefined) {
         headers = init.headers
       }
 
       // 3. Empty this’s headers’s header list.
-      this[kState].headersList = new HeadersList()
-      this[kHeaders][kHeadersList] = this[kState].headersList
+      this[kHeaders] = new Headers()
+      this[kState].headersList = this[kHeaders][kHeadersList]
```

Comment:
This doesn’t seem to be quite the same as the spec?

Reply:
`Headers` is just a wrapper around `HeadersList` so emptying the headers list is as easy as creating a new instance of headers.

`this[kState]`'s headersList has to be the same as `this[kHeaders][kHeadersList]` so we need to re-assign the value after creating a new Headers instance.

Reply:
I’m a little worried that changing the reference might have unintended side effects. Why not just empty the list as the spec says?

Reply:
Spliec(0)

Reply:
The list is a class with a symbol key that's a map but even calling `.clear()` on it or setting it to a new Map causes the tests to break. This is the only solution that works. I don't think changing the reference matters because this only happens in the constructor before we can even access the properties.

Reply:
Something is happening since the tests fail…

Reply:
Tests pass when setting it to a new Headers instance (as it currently is) but everything else causes them to fail.

Reply:
Found this issue and I'll give you a hint about it, pretty funny...

look at the changes to Headers guard...